### ISSUE TITLE
Trait HasSubscriptions | TypeError on $user->subscribedPlans()

### ISSUE THREAD

**Issue by stevensgsp** [rinvex/laravel-subscriptions#91]
`subscribedPlans()` method in `HasSubscriptions` trait returns a `/Illuminate/Database/Eloquent/Collection` instance, but the signature is 
`public function subscribedPlans(): ?PlanSubscription`

HasSubscription trait:
```
    /**
     * Get subscribed plans.
     *
     * @return \Rinvex\Subscriptions\Models\PlanSubscription|null
     */
    public function subscribedPlans(): ?PlanSubscription
    {
        $planIds = $this->subscriptions->reject->inactive()->pluck('plan_id')->unique();

        return app('rinvex.subscriptions.plan')->whereIn('id', $planIds)->get();
    }
```

**Comment by bondan23** [comment]
I think it should be return as `Collection` instead of `PlanSubscription`.

```php
public function subscribedPlans(): ?Collection
```

**Comment by stevensgsp** [comment]
That's right, specifically a `/Illuminate/Database/Eloquent/Collection` instance.

```php

use /Illuminate/Database/Eloquent/Collection;

trait HasSubscriptions
{
    ...
    public function subscribedPlans(): ?Collection
}
```

**Comment by bondan23** [comment]
> That's right, specifically a `/Illuminate/Database/Eloquent/Collection` instance.
> 
> ```
> 
> use /Illuminate/Database/Eloquent/Collection;
> 
> trait HasSubscriptions
> {
>     ...
>     public function subscribedPlans(): ?Collection
> }
> ```

yeps you can override that function and change the return to Collection instead.

**Comment by stevensgsp** [comment]
Yes, but I open the incidence so that it is corrected, since when inheriting from the Trait I cannot overwrite due to the method signature.

**Comment by Omranic** [comment]
Thank you, this is now fixed in #164
